src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.CompilationData.cs

```diff
@@ -64,92 +62,46 @@ public bool RemoveCachedSemanticModel(SyntaxTree tree)
 
             internal DeclarationAnalysisData GetOrComputeDeclarationAnalysisData(
                 SyntaxReference declaration,
-                Func<Func<DeclarationAnalysisData>, DeclarationAnalysisData> computeDeclarationAnalysisData,
+                Func<DeclarationAnalysisData> computeDeclarationAnalysisData,
                 bool cacheAnalysisData)
             {
                 if (!cacheAnalysisData)
                 {
-                    return computeDeclarationAnalysisData(_declarationAnalysisDataPool.Allocate);
+                    return computeDeclarationAnalysisData();
                 }
 
-                DeclarationAnalysisData data;
                 lock (_declarationAnalysisDataMap)
                 {
-                    if (_declarationAnalysisDataMap.TryGetValue(declaration, out data))
+                    if (_declarationAnalysisDataMap.TryGetValue(declaration, out var cachedData))
```

Comment:
>var [](start = 81, length = 3)

Please spell out the type.

Reply:
If you are planning to review this item in pull requests, please modify this line:

https://github.com/dotnet/roslyn/blob/6ac2c2708f622a451fc395cf6673cfbc9244050b/src/Compilers/.editorconfig#L6

The current statement is this is at the discretion of the implementer, which does not align with the code reviews from the compiler team.

Reply:
Changing editorconfig is not a pre-req for commenting on pull requests. The compiler team prefers that `var` be avoided in cases where the type is not obvious. This policy has been in place since the beginning of Roslyn. Shouldn't have to re-litigate this on every PR.

Reply:
I think sam's point is that this problem is avoidable if you change that.  Namely, that people will see the problem ahead of time *pre-pr*, and thus can address it.  It will mean that during the PR this conversation doesn't need to be repeated over and over again.

in a sense, this is precisely *not* re-litigating.  It's allowing the compiler to emphatically say  "that var be avoided in cases where the type is not obvious", and have that be respected so that time isn't wasted on this for compiler PRs.

Reply:
I have not contradicted that.  Nor do i think that sam was saying that.  :) The point was that if you *do* change this, it will actually make things *smoother* (both for the compiler and people making PRs) by eliminating the need for these comments entirely.

There was no claim that this was a pre-req.  Just that this sort of change would help free up time and remove this sort of noise from PRs, freeing up people to focus on more critical things.

Reply:
Sam specifically said that.

>  If you are planning to review this item in pull requests, please modify this line

Reply:
YEs.  That sounds liek a very pleasant suggestion on a way to alleviate the cost of having to review for this in future PRs.  It's not at all a pre-req.  Just something to consider as a way so that this becomes a non-issue in the future.  You are, of course, free to *not* do that (and i don't think anyone would say otherwise).  But, that just leaves us in the current position where compiler devs have to exercise mental cycles in reviews repeatedly looking for this and asking for it to be addressed, instead of having it be taken care of prior to the code ever even making it to the PR stage.

Again, if that's what you would prefer *that's totally fine*.  No one is saying you have to take this.  But, to me, it seems like a great idea simply because it both aids those making the PR (since they will know and follow the compiler rules), and it assists those reviewing things (since they don't have to spend cycles on this topic).

Reply:
> Changing editorconfig is not a pre-req for commenting on pull requests.

I'm not saying it is a pre-requisite. I'm saying it will reduce noise in the pull request and will be less of a shock for users who: 

1. Work elsewhere in dotnet/roslyn
2. Become used to the UI alerting them to use `var` everywhere
3. Write code in the compiler layer using `var`, and find no UI indication that anything is wrong
4. Submit review, blows up